File: apps/admin-v2/src/domain/repositories/availability.repository.ts
import type { AvailabilitySlot, AvailabilityBlock } from '../entities/availability';

export interface CreateBlockData {
  date: string;
  startTime?: string;
  endTime?: string;
  reason?: string;
}

export interface AvailabilityRepository {
  getSlots(): Promise<AvailabilitySlot[]>;
  updateSlots(slots: AvailabilitySlot[]): Promise<AvailabilitySlot[]>;
  getBlocks(): Promise<AvailabilityBlock[]>;
  createBlock(data: CreateBlockData): Promise<AvailabilityBlock>;
  deleteBlock(id: string): Promise<void>;
}

File: apps/admin_mobile/lib/domain/repositories/availability_repository.dart
import '../entities/availability.dart';

abstract class AvailabilityRepository {
  Future<List<AvailabilitySlot>> getSlots();
  Future<List<AvailabilitySlot>> updateSlots(List<AvailabilitySlot> slots);
  Future<List<AvailabilityBlock>> getBlocks();
  Future<AvailabilityBlock> createBlock({
    required String date,
    String? startTime,
    String? endTime,
    String? reason,
  });
  Future<void> deleteBlock(int id);
}
